app/controllers/application_controller.rb

```diff
@@ -15,12 +15,18 @@ def return_current_user_projects
     end
   end
 
+  def generate_thumbnail(path, imagefile, commit_id)
+    image = Magick::Image.read("#{path}/satellite/#{imagefile.original_filename}").first
+    image.scale(50,50).write "#{path}/thumbnails/#{commit_id}"
```

Comment:
the thumbnail size is something we might want to revisit - we'll have variable sizes of images and setting a fixed dimension might not be the best approach. ofc, it's better than what we have atm, just noting.